feat: loki.source.file file_match block (#4743)
grafana/alloy: docs/sources/reference/components/local/local.file_match.md

`local.file_match` discovers files on the local filesystem using glob patterns and the [doublestar][] library.

{{< admonition type="note" >}}
For simple use cases where you only need to discover files for a single `loki.source.file` component, consider using the built-in [`file_match`](../../loki/loki.source.file/#file_match) block in `loki.source.file` instead.
This provides better performance and simpler configuration by integrating file discovery directly into the component.

> Use `local.file_match` when you need to share discovered targets across multiple components, reuse file discovery results in different parts of your configuration, or accept targets from other discovery components (e.g., `discovery.relabel`) before file matching.

Reviewer: ```suggestion
Use `local.file_match` component when you need to share discovered targets across multiple components, reuse file discovery results in different parts of your configuration, or accept targets from other discovery components (e.g., `discovery.relabel`) before file matching.
```

It probably should be used if there's a need to do some more processing of targets it outputs, right?

`local.file_match` discovers files on the local filesystem using glob patterns and the [doublestar][] library.

{{< admonition type="note" >}}
You should always prefer using the built-in [`file_match`](../../loki/loki.source.file/#file_match) block in `loki.source.file` when possible.
This provides better performance by integrating file discovery directly into the component, eliminating the overhead of a separate component.
The performance benefit is especially noticeable in environments with many files to watch.

> Use `local.file_match` component only when you need to share discovered targets across multiple components, reuse file discovery results in different parts of your configuration, or accept targets from other discovery components (e.g., `discovery.relabel`) before file matching.

Reviewer 2: ```suggestion
Use the `local.file_match` component only when you need to share discovered targets across multiple components, reuse file discovery results in different parts of your configuration, or accept targets from other discovery components, for example,  `discovery.relabel` before file matching.
```
